django_lti_tool_provider/views.py

```diff
@@ -45,10 +46,19 @@ def post(self, request, *args, **kwargs):
         return self.process_request(request)
 
     def process_request(self, request):
+        if request.user.is_authenticated():
+            try:
+                lti_parameters = self._get_lti_parameters_from_request(request)
+                if not self._right_user(request.user, lti_parameters):
+                    _logger.debug(u"Logging out user %s in favor of new LTI session.", request.user.username)
+                    logout(request)
+            except (oauth2.Error, AttributeError):
+                # Not a new visit, or better to keep existing auth.
+                pass
```

Comment:
@e-kolpakov If a user is already authenticated, and no oauth is provided, they should stay authenticated as they are. The only time the oauth should be provided is on the initial load, right? If a user is navigating around the iFrame and clicks a link, they're not going to resend the OAuth token, are they? Or is this function not called then?

AttributeError will occur if OAuth info is missing entirely.  WrongUserError will be caught by _right_user.